ls1intum/Artemis
# `General`: Allow CSV user import by email

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Currently either the users' **login** or **registration number** has to be provided within the CSV file in order to import users to a course's group. It may happen that instructors possess only a list of the students' emails and therefore cannot import them using the CSV import functionality.

### Description

As  the **email** represent another attribute which can be used to uniquely identify a user, this can be integrated in the already implemented CSV import functionality, extending its use cases.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to Course Administration
3. Add users to one or more of the course's groups (Student, Tutors, Editors, Instructors) using the 
![image](https://user-images.githubusercontent.com/45761921/194307428-02ae6693-7e37-46e9-baa0-de6def618b4f.png)
CSV import
4. The CSV used should contain at least one row containing a user's email only

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| CourseService.java | 78% | 92% |
| ExamRegistrationService.java | 85% | 97% |
| UserService.java | 78% | 90% |

### Screenshots
